text/tls-and-alpn.rst

```diff
@@ -0,0 +1,380 @@
+the EdgeDB instances. The cert files will be named after the instance names.
+For development scenario we'll likely generate those certs for the user, while
+in production we'll "import" the certs specified by the user, by copying and
+renaming the cert and key files to the ``certs/`` folder.
+
+
+Development Scenario
+^^^^^^^^^^^^^^^^^^^^
+
+From the user's perspective, ``edgedb server init`` (without parameters we are
+introducing in this RFC later) will now automatically generate a self-signed
+certificate with a random key passphrase, and set the above config properly.
+This is considered to be the development scenario, so we try not to let the
+user do extra work here - the ``edgedb server init`` command shouldn't ask
+questions about generating the self-signed certificate. Instead, it should warn
+about the self-signed certificate, and provide suggestions for production.
+
+Question: because the certificate is self-signed, certificate verification on
+the client side would fail. Now that we know we are "in development mode"
+(using self-signed certificate), we could leverage the tooling to trust the
+certificate on the client side, by adding a boolean flag in
+``$WELLKNOWN/credentials/<instance>.json``. We could update our client bindings
+and the CLI to honor this flag, load and trust the certificate so that the user
+doesn't have to write extra code in development. Though, this works only for
+local clients - remote clients don't have access to the local file system so
+there has to be a different way for them to trust the dev server. If this
+option is good, I will expand ths paragraph with details. Alternatives:
+
+1. Automatically build a local CA, and issue certificates from that CA. The
+   clients and CLI will only need to trust the CA certificate. This doesn't
+   solve the remote client trouble though.
+2. Provide a client-side switch that the user could easily set, so that the
+   user could explicitly turn it on for development mode, and off for
+   production.
+
+
+Production Scenario
+^^^^^^^^^^^^^^^^^^^
+
+Question: do we want to manage the certificate for the user? That means to
+import the user certificate files and store locally, warn about expiry or even
+run certbot to renew some free certs (this might be too much). If yes, the
+following in this section is just one of the options.
+
+``edgedb server init`` will have additional parameters to import the TLS
+certificate and key files for the new EdgeDB instance::
+
+    --import-tls-certfile
+    --import-tls-keyfile
+    --tls-private-key-password
+
+Again the password option here yields a security concern as mentioned
+previously. We should probably follow the way PostgreSQL used to make it a
+customizable command to obtain the passphrase.
+
+For simplicity, optional TLS settings are not provided on ``server init``.
+These options will also be added to ``edgedb project init`` command.
+
+In addition, we should provide a new ``edgedb server update-tls`` command for
```

Comment:
If we want updating cli params, we should have non-TLS specific command (`edgedb server update-config` or something), and implement those parameters in it.